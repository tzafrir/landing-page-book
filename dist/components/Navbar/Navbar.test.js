'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _enzyme = require('enzyme');

var _enzymeAdapterReact = require('enzyme-adapter-react-16');

var _enzymeAdapterReact2 = _interopRequireDefault(_enzymeAdapterReact);

var _Navbar = require('./Navbar');

var _Navbar2 = _interopRequireDefault(_Navbar);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

(0, _enzyme.configure)({ adapter: new _enzymeAdapterReact2.default() });

describe('Navbar', function () {
  var props = void 0;
  var mounted = void 0;
  var navbar = function navbar() {
    if (!mounted) {
      mounted = (0, _enzyme.shallow)(_react2.default.createElement(_Navbar2.default, props));
    }
    return mounted;
  };
  describe('Navbar', function () {
    beforeEach(function () {
      props = {
        transparent: false,
        fixed: false,
        brandName: 'front10',
        brandLink: 'https://www.front10.com',
        brandLogo: '/assets/images/logo/front10.png',
        className: '',
        expand: '',
        leftItems: [],
        rightItems: [],
        children: null,
        onItemClick: function onItemClick() {}
      };
      mounted = undefined;
    });

    it('should render', function () {
      expect((0, _enzyme.shallow)(_react2.default.createElement(_Navbar2.default, props))).toMatchSnapshot();
    });

    it('Should render', function () {
      var divs = navbar().find('nav');

      expect(divs.length).toBeGreaterThan(0);
    });

    it('Collapse', function () {
      navbar().instance().toggleCollapse();

      expect(navbar().state('collapse')).toBeTruthy();
    });
  });
});