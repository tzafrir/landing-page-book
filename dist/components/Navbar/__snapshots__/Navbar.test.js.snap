// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Navbar should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar
    brandLink="https://www.front10.com"
    brandLogo="/assets/images/logo/front10.png"
    brandName="front10"
    className=""
    expand=""
    fixed={false}
    leftItems={Array []}
    onItemClick={[Function]}
    rightItems={Array []}
    transparent={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          active={false}
          ariaLabel="Toggle navigation"
          circle={false}
          className="navbar-toggler mr-3 Navbar__Toggler"
          color="light"
          disabled={false}
          loading={false}
          loadingClass="fa fa-circle-o-notch fa-spin"
          onClick={[Function]}
          outline={false}
          rounded={true}
          size=""
          style={Object {}}
          tooltip=""
          type="button"
        >
          <Icon
            className=""
            icon="fa fa-bars"
            onClick={[Function]}
            style={Object {}}
            title=""
          />
        </Button>,
        <NavbarBrand
          className="mr-auto"
          href="https://www.front10.com"
        >
          <Image
            alt="Company logo"
            border={false}
            className="d-inline-block align-top mr-1 Logo"
            imgFilter={null}
            rounded={false}
            shadow={false}
            src="/assets/images/logo/front10.png"
            tooltip=""
            width={undefined}
          />
          front10
        </NavbarBrand>,
        false,
        false,
        null,
      ],
      "className": " navbar navbar-expand Navbar navbar-down",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "ariaLabel": "Toggle navigation",
          "children": <Icon
            className=""
            icon="fa fa-bars"
            onClick={[Function]}
            style={Object {}}
            title=""
          />,
          "circle": false,
          "className": "navbar-toggler mr-3 Navbar__Toggler",
          "color": "light",
          "disabled": false,
          "loading": false,
          "loadingClass": "fa fa-circle-o-notch fa-spin",
          "onClick": [Function],
          "outline": false,
          "rounded": true,
          "size": "",
          "style": Object {},
          "tooltip": "",
          "type": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "",
            "icon": "fa fa-bars",
            "onClick": [Function],
            "style": Object {},
            "title": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Image
              alt="Company logo"
              border={false}
              className="d-inline-block align-top mr-1 Logo"
              imgFilter={null}
              rounded={false}
              shadow={false}
              src="/assets/images/logo/front10.png"
              tooltip=""
              width={undefined}
            />,
            "front10",
          ],
          "className": "mr-auto",
          "href": "https://www.front10.com",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "Company logo",
              "border": false,
              "className": "d-inline-block align-top mr-1 Logo",
              "imgFilter": null,
              "rounded": false,
              "shadow": false,
              "src": "/assets/images/logo/front10.png",
              "tooltip": "",
              "width": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "front10",
        ],
        "type": [Function],
      },
      false,
      false,
      null,
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            active={false}
            ariaLabel="Toggle navigation"
            circle={false}
            className="navbar-toggler mr-3 Navbar__Toggler"
            color="light"
            disabled={false}
            loading={false}
            loadingClass="fa fa-circle-o-notch fa-spin"
            onClick={[Function]}
            outline={false}
            rounded={true}
            size=""
            style={Object {}}
            tooltip=""
            type="button"
          >
            <Icon
              className=""
              icon="fa fa-bars"
              onClick={[Function]}
              style={Object {}}
              title=""
            />
          </Button>,
          <NavbarBrand
            className="mr-auto"
            href="https://www.front10.com"
          >
            <Image
              alt="Company logo"
              border={false}
              className="d-inline-block align-top mr-1 Logo"
              imgFilter={null}
              rounded={false}
              shadow={false}
              src="/assets/images/logo/front10.png"
              tooltip=""
              width={undefined}
            />
            front10
          </NavbarBrand>,
          false,
          false,
          null,
        ],
        "className": " navbar navbar-expand Navbar navbar-down",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "ariaLabel": "Toggle navigation",
            "children": <Icon
              className=""
              icon="fa fa-bars"
              onClick={[Function]}
              style={Object {}}
              title=""
            />,
            "circle": false,
            "className": "navbar-toggler mr-3 Navbar__Toggler",
            "color": "light",
            "disabled": false,
            "loading": false,
            "loadingClass": "fa fa-circle-o-notch fa-spin",
            "onClick": [Function],
            "outline": false,
            "rounded": true,
            "size": "",
            "style": Object {},
            "tooltip": "",
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "icon": "fa fa-bars",
              "onClick": [Function],
              "style": Object {},
              "title": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Image
                alt="Company logo"
                border={false}
                className="d-inline-block align-top mr-1 Logo"
                imgFilter={null}
                rounded={false}
                shadow={false}
                src="/assets/images/logo/front10.png"
                tooltip=""
                width={undefined}
              />,
              "front10",
            ],
            "className": "mr-auto",
            "href": "https://www.front10.com",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "Company logo",
                "border": false,
                "className": "d-inline-block align-top mr-1 Logo",
                "imgFilter": null,
                "rounded": false,
                "shadow": false,
                "src": "/assets/images/logo/front10.png",
                "tooltip": "",
                "width": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "front10",
          ],
          "type": [Function],
        },
        false,
        false,
        null,
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
